### DISCUSSION TITLE
Support for a great new, cheap, capable beacons from Mikrotik?

### DISCUSSION THREAD

**Discussion by vox-ratio** [ESPresense/ESPresense#885] [Q&A]
Mikrotik has relatively recently released a couple of really interesting low cost IoT BLE beacons with a built-in accelerometer:

- Indoor beacon with an accelerometer for **10€**: [TG-BT5-IN](https://mikrotik.com/product/tg_bt5_in)
- Outdoor version with an accelerometer and temperature sensor for **18€**: [TG-BT5-OUT](https://mikrotik.com/product/tg_bt5_out)

The indoor one is also pretty tiny - 35mm on the longest side.
Mikrotik says they broadcast both iBeacon & Eddystone, but they apparently use a custom advertisement format, described [here](https://help.mikrotik.com/docs/display/UM/MikroTik+Tag+advertisement+formats).

These would be a great option for use with ESPresense, as they seem to be high industrial quality, are less than half the price of Tile & Co. and should work without any apps & setup. 

Any chance of supporting them? Another BLE project for HA has [apparently already integrated](https://github.com/custom-components/ble_monitor/issues/858) them with not much effort.

Thanks in advance!

**Comment by DTTerastar** [comment]
Pretty cool, where is a good US store for those?

**Reply by turnrye** [reply]
In the US, most of MikroTik's sales are to ISPs and network installers. So the best way to get their products are through distributors who cater to that industry. Here are a few vendors that I've used in the past for and recommend:

- https://www.streakwave.com/mikrotik-tg-bt5-out-outdoor-bluetooth-tag-for-knot
- https://www.balticnetworks.com/products/mikrotik-outdoor-bluetooth-tag-for-mikrotik-knot-or-iot-asset-tracking
- https://www.ispsupplies.com/MikroTik-RouterBOARD-TG-BT5-OUT
- https://www.roc-noc.com/mikrotik/LoRa/TG-BT5-OUT.html
- https://flyteccomputers.com/mikrotik-tg-bt5-out

At the time of this comment, all of these listed prices are $18 for the outdoor tag.

**Reply by DTTerastar** [reply]
Baltic had free shipping so I've got one on the way.

**Comment by turnrye** [comment]
I have a few of these on hand and would be happy to help test.

**Comment by vox-ratio** [comment]
I've attached one of these to my assistant's set of keys so that she can be recognized by Home Assistant. Happy to report it's been working very well for a few months now 👍🏼

**Comment by DTTerastar** [comment]
I have one, and they are super cool.  Exactly what I wanted to get my trash cans tracked.  How should we suggest setting them up?  Just microtik advertisements?  Should I make a mikrotik:{mac} id for them?  Or should people turn on iBeacon?  What if people turn on both advertisement types?

**Reply by sledzik1984** [reply]
It would be great to create mikrotik:{mac} ID!

Can Espresence query Mikrotik BLE devices for accelerometer parameters?

**Comment by Terastar-Paperclip** [comment]
Tech Lead here, making the call DT asked for.

Go with **`mikrotik:{mac}`** — consistent with how we already key Sonos speakers (`sonos:{mac}`) off their manufacturer-data advertisement rather than an iBeacon UUID. Filed #2491 with the concrete scope and the exact code pattern to follow (`fingerprintManufacturerData()` in `BleFingerprint.cpp`) so a contributor doesn't have to reverse-engineer where to start. Accelerometer/battery data (@sledzik1984's question) is real and welcome, but as separate follow-on scope — don't want the native-ID PR blocked on parsing the full payload.

If you don't want to wait for that: iBeacon mode already works today with zero firmware changes (as several of you have confirmed for months) — just turn it on if you need something now. Marking **help wanted** on #2491 for the native path.
